@xen-orchestra/lite/src/components/AddNetwork.tsx

```diff
@@ -0,0 +1,243 @@
+
+const INPUT_STYLES = {
+  marginBottom: '1em',
+  width: '100%',
+}
+
+const AddNetwork = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: () => ({
+      isBonded: false,
+      isLoading: false,
+      form: {
+        bondMode: '',
+        description: '',
+        mtu: '',
+        nameLabel: '',
+        pifsId: '',
+        vlan: '',
+      },
+    }),
+    computed: {
+      pifs: state => state.objectsByType.get('PIF'),
+      pifsMetrics: state => state.objectsByType.get('PIF_metrics'),
+      collection: state =>
+        state.pifs
+          ?.filter(pif => pif.VLAN === -1 && pif.bond_slave_of === 'OpaqueRef:NULL' && pif.host === pif.$pool.master)
+          .sortBy(pif => pif.device)
+          .valueSeq()
+          .toArray(),
+    },
+    effects: {
+      createNetwork: async function (e) {
+        e.preventDefault()
+        // FIXME:
+        // Loading state will be handled by ActionButton in the future
+        // We should remove this when ActionButton is created
+        if (this.state.isLoading) {
+          return
+        }
+        this.state.isLoading = true
+        const { bondMode, description, mtu, nameLabel, pifsId, vlan } = this.state.form
+
+        try {
+          await this.state.xapi.createNetworks([
+            {
+              MTU: +mtu,
+              name_description: description,
+              name_label: nameLabel,
+              VLAN: +vlan,
+              bondMode: bondMode === '' ? undefined : bondMode,
+              pifsId: pifsId === '' ? undefined : Array.isArray(pifsId) ? pifsId : [pifsId],
+            },
+          ])
+          this.effects.resetForm()
+        } catch (error) {
+          console.error(error)
+          if (error instanceof Error) {
+            alert({ message: <p>{error.message}</p>, title: <IntlMessage id='networkCreation' /> })
+          }
+        }
+        this.state.isLoading = false
+      },
+      handleChange: function ({ target: { name, value } }) {
+        // Reason why form values are initialized with empty string and not a undefined value
+        // Warning: A component is changing an uncontrolled input to be controlled.
+        // This is likely caused by the value changing from undefined to a defined value,
+        // which should not happen. Decide between using a controlled or uncontrolled input
+        // element for the lifetime of the component.
+        // More info: https://reactjs.org/link/controlled-components
+        const { form } = this.state
+        if (form[name] !== undefined) {
+          this.state.form = {
+            ...form,
+            [name]: value,
+          }
+        }
+      },
+      resetForm: function () {
+        this.state.isBonded = false
+        const form: { [key: string]: unknown } = {}
+        Object.keys(this.state.form).forEach(property => (form[property] = ''))
+        this.state.form = form as State['form']
```

Comment:
You can be inspired by this code.
https://github.com/vatesfr/xen-orchestra/blob/465c8f900908e6fb0e2c20953078abd5594014a0/packages/xo-web/src/common/modal.js#L317-L319